src/miscellaneous.py

```diff
@@ -0,0 +1,154 @@
+import sys, json
```

Comment:
Can you add unit tests for this change?

Reply:
Yep I will add some unit tests later. 
And I think those tests can only be passed at local or on the server, but not on CI since we need MongoDB env setup for connecting